### PR TITLE
Improve GIF playback

### PR DESCRIPTION
**Pull request by gan-of-culture**
Addition to #919

This actually makes the GIFs look more natural in HTML preview by removing the video controls and outplaying on hover.

### CODE REVIEW

**Comment by gan-of-culture** [comment]
If a GIF enters the view it is played if it leaves the view it is paused. In my test document with 750+ GIFs this cause no lag or slow down in any other way. The observers are automatically initialized after the document has completely loaded.

**Comment by gan-of-culture** [comment]
Old:
old

New:
new

**Comment by Tyrrrz** [comment]
Can you instead have the gif play on mouse hover? Should be easier to implement too because you can actually put the JS in the handler, without doing query selector

**Comment by gan-of-culture** [comment]
Sure I can change it back, but then you have to click on every single GIF to play it. That is quite some effort if you have many GIFs^^

Personally I prefer the way how it is now, but I guess it is preference after all. I'll leave it up to you to decide :)

**Comment by Tyrrrz** [comment]
> Sure I can change it back, but then you have to click on every single GIF to play it. That is quite some effort if you have many GIFs^^
> 
> Personally I prefer the way how it is now, but I guess it is preference after all. I'll leave it up to you to decide :)

Not click, but just mouse hover. It's actually a setting in Discord itself which I use because some channels have way too many gifs.

Essentially, the gif is played only when the cursor is over it, and pauses immediately when it loses focus.

**Comment by gan-of-culture** [comment]
I have to many branches, but looks good now!

File: DiscordChatExporter.Core/Exporting/Writers/Html/MessageGroupTemplate.cshtml [hunk 1]
```diff
@@ -214,7 +214,7 @@
                         @if (embed.Video is not null && !string.IsNullOrWhiteSpace(embed.Video.Url) && embed.Thumbnail is not null)
                         {
                             <div class="chatlog__attachment">
-                                <video class="chatlog__attachment-media" poster="@await ResolveUrlAsync(embed.Thumbnail.ProxyUrl ?? embed.Thumbnail.Url)" loop="" aria-label="GIF" width="@(embed.Video.Width)" height="@(embed.Video.Height)">
+                                <video class="chatlog__attachment-media" poster="@await ResolveUrlAsync(embed.Thumbnail.ProxyUrl ?? embed.Thumbnail.Url)" loop="" onmouseover="this.play()" onmouseout="this.pause();this.currentTime=0;" aria-label="GIF" width="@(embed.Video.Width)" height="@(embed.Video.Height)">
```
Comment (Tyrrrz) [inline DiscordChatExporter.Core/Exporting/Writers/Html/MessageGroupTemplate.cshtml:217]: No need to reset the time, it's actually more convenient if it continues

```suggestion
                                <video class="chatlog__attachment-media" poster="@await ResolveUrlAsync(embed.Thumbnail.ProxyUrl ?? embed.Thumbnail.Url)" loop onmouseover="this.play()" onmouseout="this.pause()" aria-label="GIF" width="@(embed.Video.Width)" height="@(embed.Video.Height)">
```

File: .vscode/launch.json [hunk 2]
```diff
@@ -0,0 +1,24 @@
+{
```
Comment (Tyrrrz) [inline .vscode/launch.json]: Looks like you need to add `.vscode` to `gitignore` in this PR too

File: DiscordChatExporter.Core/Exporting/Writers/Html/MessageGroupTemplate.cshtml [hunk 3]
```diff
@@ -214,7 +214,7 @@
                         @if (embed.Video is not null && !string.IsNullOrWhiteSpace(embed.Video.Url) && embed.Thumbnail is not null)
                         {
                             <div class="chatlog__attachment">
-                                <video class="chatlog__attachment-media" poster="@await ResolveUrlAsync(embed.Thumbnail.ProxyUrl ?? embed.Thumbnail.Url)" loop="" controls="" aria-label="GIF" width="@(embed.Video.Width)" height="@(embed.Video.Height)">
+                                <video class="chatlog__attachment-media" poster="@await ResolveUrlAsync(embed.Thumbnail.ProxyUrl ?? embed.Thumbnail.Url)" loop onmouseover="this.play()" onmouseout="this.pause();" aria-label="GIF" width="@(embed.Video.Width)" height="@(embed.Video.Height)">
```
Comment (Tyrrrz) [inline DiscordChatExporter.Core/Exporting/Writers/Html/MessageGroupTemplate.cshtml:217]:
```suggestion
                                <video class="chatlog__attachment-media" poster="@await ResolveUrlAsync(embed.Thumbnail.ProxyUrl ?? embed.Thumbnail.Url)" loop onmouseover="this.play()" onmouseout="this.pause()"  width="@embed.Video.Width" height="@embed.Video.Height">
```

File: DiscordChatExporter.Core/Exporting/Writers/Html/MessageGroupTemplate.cshtml [hunk 4]
```diff
@@ -214,7 +214,7 @@
                         @if (embed.Video is not null && !string.IsNullOrWhiteSpace(embed.Video.Url) && embed.Thumbnail is not null)
                         {
                             <div class="chatlog__attachment">
-                                <video class="chatlog__attachment-media" poster="@await ResolveUrlAsync(embed.Thumbnail.ProxyUrl ?? embed.Thumbnail.Url)" loop="" controls="" aria-label="GIF" width="@(embed.Video.Width)" height="@(embed.Video.Height)">
+                                <video class="chatlog__attachment-media" poster="@await ResolveUrlAsync(embed.Thumbnail.ProxyUrl ?? embed.Thumbnail.Url)" loop onmouseover="this.play()" onmouseout="this.pause()"  width="@embed.Video.Width" height="@embed.Video.Height">
```
Comment (Tyrrrz) [inline DiscordChatExporter.Core/Exporting/Writers/Html/MessageGroupTemplate.cshtml:217]:
```suggestion
                                <video class="chatlog__attachment-media" poster="@await ResolveUrlAsync(embed.Thumbnail.ProxyUrl ?? embed.Thumbnail.Url)" loop onmouseover="this.play()" onmouseout="this.pause()" width="@embed.Video.Width" height="@embed.Video.Height">
```

**Comment by Tyrrrz** [comment]
Thanks!

**Pull request merged**